### ISSUE TITLE
Promise not handled correctly on empty url values

### ISSUE THREAD

**Issue by MikeBroders** [graysonhicks/gatsby-plugin-remote-images#86]
hey,

thanks for your effort to provide this plugin!

there seems to be an issue while handling the image nodes in the createResolvers function.

especially in the condition if the url is a string.

because the result of:

```
const fileNodeMap = await cache.get(
              getCacheKeyForNodeId(source.id)
            );
```

could be empty, a check of the fileNodeMap would be needed before this:

`return context.nodeModel.getNodeById({ id: fileNodeMap[name] });`

it throws a resolve error on dev & build if the url for the image was empty.

many thanks
mike

**Comment by graysonhicks** [comment]
Ah, yea, looks like there is a check in one usage of that, but not there.  I'll add it.

**Comment by graysonhicks** [comment]
Added check [here](https://github.com/graysonhicks/gatsby-plugin-remote-images/blob/master/src/gatsby-node.js#L236) and tested in a demo site successfully.  Published with 3.3.1, thanks!
